models/users.js

```diff
@@ -234,6 +234,24 @@ module.exports = {
     return userSecret.rows[0].secret
   },
 
+  getTrueUserEmailByEmail: async function (email) {
+    const userExists = await this.getUserIdByEmail(email)
+    if (!userExists) {
+      return null
+    }
+    const getUserEmail = {
+      text: 'SELECT u2.email FROM users u1 ' +
+      'JOIN users u2 ON u1.mergeduserid = u2.id ' +
+      'WHERE u1.email = $1',
+      values: [email]
+    }
+    const userEmail = await db.query(getUserEmail)
+    if (userEmail.rows.length === 0) {
+      return email
+    }
+    return userEmail.rows[0].email
+  },
+
   getUserEmailBySecret: async function (secret) {
     const getUserSecret = {
```

Comment:
This duplication of logic is getting ridiculous.

We should have one function that gets ALL user data by `userId` (so we don't have 5 functions that all do the same sql search). Code that needs user data should either have `userId` or `getTrueUserByEmail` or `bySecret`. Then we can delete everything else.

The amount of speed savings we get from fetching specific columns rather than the entire table + friendship joined, is negligible vs the fact we have to write new model code everytime we want something from the database.

Does that make sense?

At the same time, you did good keeping this MR changeset focused, so I don't want to block you from merging.

Reply:
Ill make a separate PR for cleaning duplicate code. But what you are suggesting is instead of making more db calls, we should use a general `getUserData` and parse that general data for whatever specific thing we need in the services. For example, `getMergedUserId` can be removed because we will simply call `getUserData` and get the mergedUserId from the userData. Is this correct?

Reply:
Yes, exactly! Looking forward to much cleaner code haha